tests/pos/z1720.scala

```diff
@@ -3,7 +3,7 @@ package test
 class Thing {
   def info: Info[this.type] = InfoRepository.getInfo(this)
   def info2: Info[this.type] = {
-    def self: this.type = this
+    val self: this.type = this
```

Comment:
So it seems that `this.type` should be as realizable as `this`, no?

Reply:
Yes, the change to `val` is no longer needed.

Reply:
Yes you are right. I added a further tweak to support this in 8389bea